package PS;
import "proto/config.proto";

message SGDProgress {
  repeated double objective = 1;
  optional uint64 num_examples_processed = 12;
  repeated double accuracy = 13;
  repeated double auc = 14;
}

message SGDCall {
  enum Command {
    UPDATE_MODEL = 1;
    REPORT_PROGRESS = 2;
    SAVE_MODEL = 3;  // save w
    RECOVER = 4;
    COMPUTE_VALIDATION_AUC = 5;
  }
  required Command cmd = 1;
  optional DataConfig data = 2;
  optional int32 report_interval = 4;
  // optional PbRange key = 3;
}
